code/game/machinery/lightswitch.dm

```diff
@@ -87,9 +87,10 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/light_switch, 26)
 	area.lightswitch = status
 	area.update_appearance()
 
-	for(var/obj/machinery/light_switch/light_switch in area)
-		light_switch.update_appearance()
-		SEND_SIGNAL(light_switch, COMSIG_LIGHT_SWITCH_SET, status)
+	for(var/turf/area_turf as anything in area.get_contained_turfs())
+		for(var/obj/machinery/light_switch/light_switch in area_turf)
+			light_switch.update_appearance()
+			SEND_SIGNAL(light_switch, COMSIG_LIGHT_SWITCH_SET, status)
```

Comment:
might actually be faster to just filter for switches in the machines list, worth profiling? how much does this actually cost

Reply:
![image](https://github.com/tgstation/tgstation/assets/51863163/e0d0fee0-1f9e-45ce-a53a-81fde1e64c53)

Reply:
![image](https://github.com/tgstation/tgstation/assets/51863163/1d9adb25-f648-4f75-87e2-65a40df00106)
![image](https://github.com/tgstation/tgstation/assets/51863163/c7b52f48-7bb0-4a11-b172-d527986ee58b)
![image](https://github.com/tgstation/tgstation/assets/51863163/41c8786b-2793-45c9-9e73-1781df612c06)